### PR TITLE
Benchmarks modified rhino

### PR DESCRIPTION
**Pull request by acepace**
Added a set of State store test and benchmarks.

Tests - `StateStoreTests`
These are a pair of _simple_ tests for comparing states with identical and different JS variables at top thread scope.

Benchmarks - `SnapshotSizeBenchmark`
This is a pair of simple tests to check for growth of a snapshot over different runs with more variables.
The tests contain basic size benchmarking (min,max,avg) and a warning for identical snapshot sizes between iterations, which **indicates a bug**. This warning is currently triggered for saving an object context.

Note: the `SnapshotSizeBenchmark` is saved under `il.ac.bgu.cs.bp.bpjs.analysis` because it uses a protected method rather than under `mains`

### CODE REVIEW

**Comment by acepace** [comment]
Note, failing tests are all tests where the Rhino interpreter threw an exception. These tests (without commit 90bd4effa9fdac43966ea3e91d663983e60e95cb ) work perfectly fine on the develop branch.

File: src/test/java/il/ac/bgu/cs/bp/bpjs/analysis/StateStoreTests.java [hunk 1]
```diff
@@ -149,7 +151,7 @@ private void TestEqualJSVar(VisitedStateStore storeToUse) throws Exception {
         //now we want to compare specific
         BProgramSyncSnapshot state1 = snapshots.get(1).getSystemState();
         BProgramSyncSnapshot state2 = snapshots.get(2).getSystemState();
-        assertNotEquals(state1,state2);
+        assertEquals(state1, state2);
```
Comment (michbarsinai) [inline src/test/java/il/ac/bgu/cs/bp/bpjs/analysis/StateStoreTests.java]: Why was this changed from `assertNotEquals` to` assertEquals`?

Comment (acepace) [reply src/test/java/il/ac/bgu/cs/bp/bpjs/analysis/StateStoreTests.java]: Note that in this function, i doesn't increment, so the state should be identical to the one 4 syncs back.

However, this is actually testing two things at once (snapshot and store) and I'll make it clearer what we're actually testing.

**Comment by acepace** [comment]
I explained your comment + added the more complicated scope tests.
This falls a bit outside the benchmark branch but this is easier than tracking two branches of tests.
